### ISSUE TITLE
Add a RunAsGroup field to SecurityContext

### ISSUE THREAD

**Issue by yuvipanda** [kubernetes/kubernetes#13868]
We want to allow containers to be able to read/write from NFS (via hostPath, rather than as mounts themselves, due to various reasons). This also allows kubernetes to co-exist with our current 'cluster' (running... GridEngine). 

For this to work we  need to be able to control the uid, gid and groups as which the container runs. `RunAsUser` already exists, so this would just add `RunAsGroup` and perhaps `RunWithSupplementaryGroups` or something similar.

I'm aware that #7893 is doing a more thorough overhaul of this, but am filing this anyway because:
1. I don't think that patch is going to land before 1.1 (or anytime soon after)
2. Is a lot more comprehensive than what we need atm

So hopefully just adding these two to SecurityContext doesn't complicate the b/c story further.

**Comment by davidopp** [comment]
@erictune @pmorie

**Comment by pweil-** [comment]
related: https://github.com/kubernetes/kubernetes/pull/12823

**Comment by pmorie** [comment]
@yuvipanda I am planning to implement this exact thing after #12823 lands, which I am hoping will be very shortly.

**Comment by yuvipanda** [comment]
#12823 seems like a fairly complex patch that is also at least somewhat controversial, while this would be much more simpler and also probably more uncontroversial (not fully sure about that).

**Comment by pmorie** [comment]
@yuvipanda I think we will try to land #13939 and continue working on the remaining aspects of #12823.  After #13939 lands, we will begin working on pod-level supplemental group.

**Comment by erictune** [comment]
@yuvipanda did the merging of #12823 and #13939 make this PR obsolete for you?

**Comment by pmorie** [comment]
This should be obsolete now, but I'll wait for @yuvipanda to weigh in

**Comment by vpal** [comment]
Are there any plans to implement the RunAsGroup besides the SupplementalGroups option?
(I see that SupplementalGroups is available in the current tree, but I was not able to get it working with 1.1.2 though, but haven't spent a lot of time with it yet.)

It really does not seem to make any sense to me to be forced to run as gid 0 and only be able to provide the user a container should run with.
So I think it would be a very beneficial feature to have an option for both controlling the primary and supplementary groups.
It would definitely also increase security.

I just started to work with Kubernetes and this was the first problem I ran into very quickly and would be very glad to see this in the near future.
Currently I see no other way than creating my own images to specify both the user and the group instead of just specifying this in the manifest.
For community or Google provided images I'm stuck with someuser:root.

Docker supports providing the user in a uid:gid form, would be also enough to allow that as a parameter for RunAsUser.
